pkg/block/index.go

```diff
@@ -559,9 +559,9 @@ func sanitizeChunkSequence(chks []chunks.Meta, mint int64, maxt int64, ignoreChk
 	var last *chunks.Meta
 
 OUTER:
-	for _, c := range chks {
+	for i := range chks {
```

Comment:
Very subtle.  So we are remembering a pointer to the chunk `c` from the last iteration to compare it against the chunk in the current iteration.  However, when we use `for index, value := range slice` the `value` is not a pointer into the slice.  In fact its a new variable the current item of the slice is copied into.  Which means our pointer based comparisons are broken -- they always compare the current chunk to itself as the address of the variable `c` doesn't change throughout the loop.

Using just a slice index here allows us to correctly store a pointer to the item of the slice from the last iteration and compare that to the chunk in the current iteration.  Otherwise, this code was removing all chunks in the series other than the first one.

Reply:
Let's document this :)

Reply:
Where is the right place to do so?  Glad to do it.

Reply:
Added comments in the code.  If that's not the best place, let me know.

Reply:
amazing, nice catch!

Reply:
More like, why did the repair just lose all the data in by blocks?

Reply:
Very nice catch, the comment makes it clear when we read this in 3 months again :) :+1: